src/source/Crypto/Dtls_openssl.c

```diff
@@ -592,6 +757,7 @@ STATUS dtlsSessionIsInitFinished(PDtlsSession pDtlsSession, PBOOL pIsConnected)
 
     CHK(pDtlsSession != NULL && pIsConnected != NULL, STATUS_NULL_ARG);
```

Comment:
releasing unacquired session. Since acquire has a null check internally, you can just place it before the null check to avoid this happening. It's happening in several functions right now.

Reply:
We're missing basic invalid unit tests like passing in null values, otherwise this should have been found by CI.

Reply:
I will add some unit tests. But `releaseDtlsSession` also has an internal NULL check. So we are neither acquiring nor releasing here, so we are safe. But, I will move the acquire before the NULL check in all functions to avoid any future issues due to code changes.

Reply:
That is correct. Misunderstood the context.